script.js

```diff
@@ -1,4 +1,317 @@
+// Global Variables
+var winCount = 0;
+var loseCount = 0;
+var drawCount = 0;
+var totalGames = 0;
+var currentGameMode = "waiting for user name";
+var username = "";
+var scissors = "scissors";
+var rock = "rock";
+var paper = "paper";
+var goodJob = "";
+var rScissors = "reversed scissors";
+var rRock = "reversed rock";
+var rPaper = "reversed paper";
+var myOutputValue = "";
+var rScissors = "reversed scissors";
+var rRock = "reversed rock";
+var rPaper = "reversed paper";
+var koreanGameStart = "waiting for winner";
+
+// Roll a random number between 1 and user chosen number.
+var rollNumber = function (max) {
+  var randomDecimal = Math.random() * max;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  return randomInteger;
+};
+// Function to determine what the computer will choose.
+var generateResult = function () {
+  var output = "try again";
+  var totalResults = 3;
+  var randomResult = rollNumber(totalResults);
+  if (randomResult == 1) {
+    output = "paper";
+  }
+  if (randomResult == 2) {
+    output = "scissors";
+  }
+  if (randomResult == 3) {
+    output = "rock";
+  }
+  return output;
+};
+
+// Choice of Game Mode
+var gameMode = function (input) {
+  if (input == "scissors, paper, stone") {
+    currentGameMode = "scissors, paper, stone"; // normal SPS
+  }
+  if (input == "reverse SPS") {
+    currentGameMode = "reverse SPS"; // SPS where the rules are reversed
+  }
+  if (input == "reverse reverse SPS") {
+    currentGameMode = "scissors, paper, stone"; // reverse reverse SPS so it's normal SPS
+  }
+  if (input == "Korean SPS") {
+    currentGameMode = "Korean SPS"; // korean SPS
+  }
+  if (input == "Computer") {
+    currentGameMode = "Computer";
+  }
+  return "you have chosen " + currentGameMode;
+};
+
+// Reversed Game Mode
+var gameResultRSPS = function (input, enemyOutput) {
+  if ((input == scissors || input == rScissors) && enemyOutput == rock) {
+    myOutputValue = "you won " + message(username, winCount, loseCount);
+    winCount = winCount + 1;
+  }
+  if ((input == scissors || input == rScissors) && enemyOutput == scissors) {
+    myOutputValue = "draw! " + message(username, winCount, loseCount);
+    drawCount = drawCount + 1;
+  }
```

Comment:
all draw scenarios can be simplified to input== enemyOutput, though simplifying this for the reverse case may be trickier.
maybe a input==enemyOutput || input.slice(1) == enemyOutput